staging/src/k8s.io/apiserver/pkg/endpoints/filters/apf_classification.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"net/http"
+
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/util/flowcontrol/classification"
+)
+
+func WithPriorityAndFairnessClassification(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		r = r.WithContext(request.WithPriorityAndFairnessClassification(ctx, &classification.PriorityAndFairnessClassification{}))
```

Comment:
Is there a reason not to build this into the main APF handler?

Reply:
Correct me if I'm wrong, but I think that if we want to extract this from context in httplog handler, then this field must be added to context before we enter httplog's handler (as r.WithContext returns only a copy of request so the original request is not changed). If we set this only in the main APF handler, then the the httplog handler (which wraps APF handler) will never see the context change.

Similar approach is used for audit-id -- it is set before other audit handlers only to make it visible in WithLogging.

Reply:
An alternative approach would be to use httplog's "addedInfo" feature and in APF's handler extract logging from context and add priorityLevel to addedInfo. In the current form "addedInfo" is simply a string you can append to, but technically we can make it a key=value map and finally have similar readability as we have in this approach.

WDYT?

Reply:
oh yes, `WithLogging` won't have access to the copied request context

Reply:
> An alternative approach would be to use httplog's "addedInfo" feature and in APF's handler extract logging from context and add priorityLevel to addedInfo

I think we already use `addedInfo`  - https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filterlatency/filterlatency.go#L66-L68

It seems to be a good alternative.

Reply:
The `addedInfo` in the current form is quite hard to read it prints as addedInfo="\nslowFilter=200ms\npriorityLevel=system", but we may want to solve readability issue anyway (by e.g. making it key=value).

I prefer sticking to the current approach as it's already done this way and I'm not sure I see any real value from migrating to that approach right now, but if we find a good reason to change approach I can do that as well.

Reply:
if we are using `httplog.AddInfof`, do we still need the filter `WithPriorityAndFairnessClassification`?

Reply:
it seems to be a runtime optimization if we can do without a new association in the request context. on the other hand, I am OK having `WithPriorityAndFairnessClassification`.